e2e/msa/msaKeyManagement.test.ts

```diff
@@ -170,6 +174,41 @@ describe('MSA Key management', function () {
       // Cleanup
       await assert.doesNotReject(ExtrinsicHelper.deletePublicKey(keys, getUnifiedPublicKey(thirdKey)).signAndSend());
     });
+
+    it('adds a new public key for free if there is just one and its before the expiration block', async function () {
+      // Set up
+      const currentBlock = await getBlockNumber();
+      setFreeKeyAddExpirationBLock(getSudo().keys, currentBlock + 30);
+      // set a completely new msa id up so it doesn't conflict with anything else.
+      const some_key = await createAndFundKeypair(fundingSource, 5n * CENTS);
+      const { target } = await ExtrinsicHelper.createMsa(some_key).signAndSend();
+      const someMsaId = target!.data.msaId;
+
+      const second_key = createKeys();
+      const newPayload = await generateAddKeyPayload({
+        ...defaultPayload,
+        newPublicKey: getUnifiedPublicKey(second_key),
+        msaId: someMsaId,
+      });
+      addKeyData = ExtrinsicHelper.api.registry.createType('PalletMsaAddKeyData', newPayload);
+      ownerSig = signPayloadSr25519(some_key, addKeyData);
+      newSig = signPayloadSr25519(second_key, addKeyData);
+
+      // act
+      const op = ExtrinsicHelper.addPublicKeyToMsa(some_key, ownerSig, newSig, newPayload);
+      const { target: event, eventMap: chainEvents } = await op.fundAndSend(fundingSource);
+
+      // assert
+      assert.notEqual(event, undefined, 'should have added public key');
+      assert.notEqual(
+        chainEvents['system.ExtrinsicSuccess'],
+        undefined,
+        'should have returned an ExtrinsicSuccess event'
+      );
+      assert.equal(chainEvents['balances.Withdraw'], undefined, 'should not have withdrawn a balance');
```

Comment:
would it be more correct to look for the event `transactionPayment.TransactionFeePaid`?

Reply:
I thought so too, but that event isn't in the event list.

Reply:
For some reason I was not seeing that event before, but added, checks for both now.